decoder/src/elf2table/symbol.rs

```diff
@@ -38,7 +41,11 @@ pub(super) enum SymbolTag {
 
 impl Symbol {
     pub fn demangle(raw: &str) -> anyhow::Result<Self> {
-        serde_json::from_str(raw)
+        let mut raw = Cow::from(raw);
+        if let Some(s) = raw.strip_prefix("__defmt_") {
```

Comment:
This string exists twice (once here, once in the macro), and it doesn't say a lot about the encoding used.

* Perhaps it should be `__defmt_hex_`, allowing for some future `__defmt_base64_` or similar.
* Is there somewhere it can be defined once and imported in the other place? If not, we should document the canonical form in a third place (like the book) and have the other two places refer to it.

Reply:
> Perhaps it should be __defmt_hex_, allowing for some future __defmt_base64_ or similar.

changed!

> Is there somewhere it can be defined once and imported in the other place? If not, we should document the canonical form in a third place (like the book) and have the other two places refer to it.

there's no crate that's a shared dep of the macro and the decoder. I've added a section to the book.

Reply:
> there's no crate that's a shared dep of the macro and the decoder. I've added a section to the book.

There is! It's the `defmt-parser`. I think it makes sense to put the magic string there.

Reply:
Huh.

```console
➜  defmt git:(main) cargo tree -p defmt-macros | grep parser
├── defmt-parser v0.3.4 (/Users/jonathan/Documents/knurling-rs/defmt/parser)

➜  defmt git:(main) cargo tree -p defmt-decoder | grep parser
├── defmt-parser v0.3.4 (/Users/jonathan/Documents/knurling-rs/defmt/parser)
```